src/Windowing/Extensions/Silk.NET.Windowing.Extensions/WindowManager.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using Silk.NET.Core.Platform;
+using Silk.NET.Windowing.Common;
+
+namespace Silk.NET.Windowing.Extensions
+{
+    /// <summary>
+    /// Contains APIs for managing multiple windows within one render loop.
+    /// </summary>
+    public class WindowManager : IDisposable
+    {
+        /// <summary>
+        /// The windows managed by this instance.
+        /// </summary>
+        // ReSharper disable once ReturnTypeCanBeEnumerable.Global
+        public ICollection<IWindow> Windows { get; } = new List<IWindow>();
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this manager's render loop is active.
+        /// </summary>
+        public bool IsRunning { get; set; }
+
+        /// <summary>
+        /// Adds a window to this manager.
+        /// </summary>
+        /// <param name="window">The window to add.</param>
+        public void AddWindow(IWindow window) => Windows.Add(window);
+
+        /// <summary>
+        /// Removes the given window from this manager.
+        /// </summary>
+        /// <param name="window">The window to remove.</param>
+        /// <returns>
+        /// Whether the removal was successful or not. One reason why this might return false is that the given window
+        /// isn't being managed by this manager.
+        /// </returns>
+        public bool RemoveWindow(IWindow window) => Windows.Remove(window);
+
+        /// <summary>
+        /// Creates a window and adds it to this manager.
+        /// </summary>
+        /// <param name="opts">The options to initialize the new window with.</param>
+        /// <param name="platform">
+        /// The windowing platform to use. If null, the current windowing platform will be
+        /// fetched using the <see cref="SilkManager"/>.
+        /// </param>
+        public void CreateWindow(WindowOptions opts, IWindowPlatform platform = null)
+            => AddWindow((platform ?? SilkManager.Get<IWindowPlatform>()).GetWindow(opts));
+
+        /// <summary>
+        /// Executes a render loop encompassing all windows within this manager.
+        /// </summary>
+        /// <param name="open">Indicates whether the window manager should open the child windows.</param>
+        public void Run(bool open = true)
+        {
+            IsRunning = true;
+            if (open)
+            {
+                foreach (var window in Windows)
+                {
+                    window.Open();
+                }
+            }
+
+            while (IsRunning)
+            {
+                foreach (var window in Windows)
+                {
+                    if (window.IsClosing)
+                    {
+                        Windows.Remove(window);
```

Comment:
I think this will cause a "Collection was modified" type of exception since you are modifying the Windows list in the middle of the iteration

Reply:
Fixed, introduced locks on all of the modifying APIs. Please approve & merge this PR if you have no further comments.